unattended_installer/install_functions/opendistro/common.sh

```diff
@@ -274,107 +274,86 @@ function restoreWazuhrepo() {
 
 function rollBack() {
 
-    if [ -z "${uninstall}" ]; then
-        logger "Cleaning the installation."
-    fi
+    componentList=("${wazuhinstalled}" "${filebeatinstalled}" "${elasticsearchinstalled}" "${kibanainstalled}")
 
-    if [ -f "/etc/yum.repos.d/wazuh.repo" ]; then
-        eval "rm /etc/yum.repos.d/wazuh.repo"
-    elif [ -f "/etc/zypp/repos.d/wazuh.repo" ]; then
-        eval "rm /etc/zypp/repos.d/wazuh.repo"
-    elif [ -f "/etc/apt/sources.list.d/wazuh.list" ]; then
-        eval "rm /etc/apt/sources.list.d/wazuh.list"
-    fi
+    logger "Analyzing components to uninstall and clean."
 
-    if [[ -n "${wazuhinstalled}" && ( -n "${wazuh}" || -n "${AIO}" || -n "${uninstall}" ) ]];then
-        logger -w "Removing the Wazuh manager."
-        if [ "${sys_type}" == "yum" ]; then
-            eval "yum remove wazuh-manager -y ${debug}"
-        elif [ "${sys_type}" == "zypper" ]; then
-            eval "zypper -n remove wazuh-manager ${debug}"
-            eval "rm -f /etc/init.d/wazuh-manager ${debug}"
-        elif [ "${sys_type}" == "apt-get" ]; then
-            eval "apt remove --purge wazuh-manager -y ${debug}"
-        fi 
+    # Uninstall case!
```

Comment:
Why the `!`?
```suggestion
    # Uninstall case
```

Reply:
Done
https://github.com/wazuh/wazuh-packages/pull/1211/commits/c3fc83525f9efa17705f53a3ff6e1f8d681d1d49